tests/Aspire.Hosting.Oracle.Tests/OracleFunctionalTests.cs

```diff
@@ -0,0 +1,377 @@
+                {
+                    var hb = Host.CreateApplicationBuilder();
+
+                    hb.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
+                    {
+                        [$"ConnectionStrings:{db2.Resource.Name}"] = await db2.Resource.ConnectionStringExpression.GetValueAsync(default)
+                    });
+
+                    hb.AddOracleDatabaseDbContext<TestDbContext>(db2.Resource.Name);
+
+                    using (var host = hb.Build())
+                    {
+                        await host.StartAsync();
+
+                        // Wait until the database is available
+                        await pipeline.ExecuteAsync(async token =>
+                        {
+                            var dbContext = host.Services.GetRequiredService<TestDbContext>();
+                            var databaseCreator = (RelationalDatabaseCreator)dbContext.Database.GetService<IDatabaseCreator>();
+                            Assert.True(await databaseCreator.CanConnectAsync(token));
+                        });
+
+                        await pipeline.ExecuteAsync(async token =>
+                        {
+                            using var dbContext = host.Services.GetRequiredService<TestDbContext>();
+                            var brands = await dbContext.Cars.ToListAsync(cancellationToken: token);
+                            Assert.Single(brands);
+                        }, cts.Token);
+
+                        await app.StopAsync();
+
+                        // Wait for the database to not be available before attempting to clean the volume.
+
+                        await pipeline.ExecuteAsync(async token =>
+                        {
+                            var dbContext = host.Services.GetRequiredService<TestDbContext>();
+                            var databaseCreator = (RelationalDatabaseCreator)dbContext.Database.GetService<IDatabaseCreator>();
+                            Assert.False(await databaseCreator.CanConnectAsync(token));
+                        }, cts.Token);
+                    }
+                }
+                finally
+                {
+                    // Stops the container, or the Volume/mount would still be in use
+                    await app.StopAsync();
+                }
+            }
+
+        }
+        finally
+        {
+            if (volumeName is not null)
+            {
+                DockerUtils.AttemptDeleteDockerVolume(volumeName);
+            }
+
+            if (bindMountPath is not null)
+            {
+                try
+                {
+                    Directory.Delete(bindMountPath, true);
+                }
+                catch
+                {
+                    // Don't fail test if we can't clean the temporary folder
+                }
+            }
+        }
+    }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false, Skip = "Scripts are not executed when in the /setup folder")]
```

Comment:
Is there an issue for this?

Reply:
https://github.com/dotnet/aspire/issues/5044